tcf_website/views/search.py

```diff
@@ -50,14 +61,24 @@ def search(request):
     # Save filters to session
     request.session["search_filters"] = filters
 
-    if query:
+    # Create a cache key from the query and filters
+    cache_key = f"search_{query}_{hash(frozenset(str(filters.items())))}"
+    cached_results = cache.get(cache_key)
```

Comment:
This just makes me nervous. Is there really no way to cache? Is below not what we're looking for?

https://docs.djangoproject.com/en/5.1/topics/cache/

```python
from django.views.decorators.cache import cache_page

@cache_page(60 * 15)
def my_view(request): ...
```